src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -354,6 +360,289 @@ def _get_a3_mega_blueprint_dependencies(
+            "autoscaling_total_min_nodes": autoscaling_total_min_nodes,
+            "initial_node_count": num_nodes,
+            "spot": spot,
+            "master_authorized_networks": [{
+                "cidr_block": auth_cidr,
+                "display_name": "kubectl-access-network",
+            }],
+            "system_node_pool_node_count": [{
+                "total_min_nodes": system_node_pool_min_node_count,
+                "total_max_nodes": 1000,
+            }],
+            "maintenance_exclusions": [{
+                "name": "no-minor-or-node-upgrades-indefinite",
+                "start_time": "2024-12-01T00:00:00Z",
+                "end_time": "2025-12-22T00:00:00Z",
+                "exclusion_scope": "NO_MINOR_OR_NODE_UPGRADES",
+            }],
+            "additional_networks": (
+                f"$(concat([{{network={cluster_name}-a3u-net-1.network_name,"
+                f" subnetwork={cluster_name}-a3u-net-1.subnetwork_name,"
+                f' subnetwork_project="{project_id}", nic_type="GVNIC",'
+                " queue_count=null, network_ip=null, stack_type=null,"
+                " access_config=[{nat_ip=null, public_ptr_domain_name=null,"
+                " network_tier=null}], ipv6_access_config=[],"
+                " alias_ip_range=[]}],"
+                f" {cluster_name}-a3u-rdma-net.subnetwork_interfaces_gke))"
+            ),
+        },
+        outputs=["instructions"],
+    )
+    gpu_pool_id = f"{cluster_name}-a3u-ultragpu-pool"
+    gpu_pool = DeploymentModule(
+        id=gpu_pool_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/compute/gke-node-pool?ref=e0c690b",
+        use=[cluster_id],
+        settings={
+            "machine_type": "a3-ultragpu-8g",
+            "auto_upgrade": True,
+            "zones": [zone],
+            "disk_type": "hyperdisk-balanced",
+            "disk_size_gb": a3ultra_node_pool_disk_size_gb,
+            "static_node_count": static_node_count,
+            "guest_accelerator": [{
+                "type": "nvidia-h200-141gb",
+                "count": 8,
+                "gpu_driver_installation_config": {
+                    "gpu_driver_version": "LATEST"
+                },
+            }],
+            "additional_networks": (
+                f"$(concat([{{network={cluster_name}-a3u-net-1.network_name,"
+                f" subnetwork={cluster_name}-a3u-net-1.subnetwork_name,"
+                f' subnetwork_project="{project_id}", nic_type="GVNIC",'
+                " queue_count=null, network_ip=null, stack_type=null,"
+                " access_config=[{nat_ip=null, public_ptr_domain_name=null,"
+                " network_tier=null}], ipv6_access_config=[],"
+                " alias_ip_range=[]}],"
+                f" {cluster_name}-a3u-rdma-net.subnetwork_interfaces_gke))"
+            ),
+        },
+        outputs=["instructions"],
+    )
+    if extended_reservation is not None:
```

Comment:
I think you can refactor in your code or [here](https://github.com/AI-Hypercomputer/xpk/blob/8e2eb525a481308a43485486de6953feaf36d7c6/src/xpk/core/blueprint/blueprint_generator.py#L150) to have a uniform reservation block implementation. You can make it a function.